pandas/core/dtypes/missing.py

```diff
@@ -124,11 +128,14 @@ def _use_inf_as_na(key):
 
 
 def _isna_ndarraylike(obj):
+    from ..arrays import ExtensionArray
 
     values = getattr(obj, 'values', obj)
     dtype = values.dtype
 
-    if is_string_dtype(dtype):
+    if isinstance(values, ExtensionArray):
```

Comment:
Should be able to replace this with `is_extension_array_dtype`

Reply:
This will break if the extension array implements a `.values` (due to two lines above). Eg if you would use that name instead of `.data`. I won't find that the best name for it, so ideally an ExtensionArray author does not do that, but we maybe shouldn't assume that? 
(or either document in the interface that that attribute should not be present)

Reply:
Another thing: won't this catch Categorical? (because there is still handling for categoricals some lines below, which might be removable)